sigstore-java/src/main/java/dev/sigstore/encryption/Keys.java

```diff
@@ -62,10 +63,15 @@ public static PublicKey parsePublicKey(byte[] keyBytes)
     PemReader pemReader =
         new PemReader(
             new InputStreamReader(new ByteArrayInputStream(keyBytes), StandardCharsets.UTF_8));
-    PemObject section = pemReader.readPemObject();
-    if (pemReader.readPemObject() != null) {
-      throw new InvalidKeySpecException(
-          "sigstore public keys must be only a single PEM encoded public key");
+    PemObject section = null;
+    try {
+      section = pemReader.readPemObject();
+      if (pemReader.readPemObject() != null) {
+        throw new InvalidKeySpecException(
+            "sigstore public keys must be only a single PEM encoded public key");
+      }
+    } catch (DecoderException e) {
+      throw new InvalidKeySpecException("Invalid key");
```

Comment:
```suggestion
      throw new InvalidKeySpecException("Invalid key, could not parse PEM section");
```

Reply:
Thanks for the comments. Changed accordingly.